### DISCUSSION TITLE
'SVG' cannot be used as a JSX component.

### DISCUSSION THREAD

**Discussion by namansamra** [gilbarbara/react-inlinesvg#186] [Q&A] [answered]
I am using this library for years but suddenly in one project this error occurs in typescript that " 'SVG' cannot be used as a JSX component ". I have tried but not able to resolve this. Please help if any one had faced the same issue.
<img width="801" alt="Screenshot 2022-04-27 at 12 34 44 PM" src="https://user-images.githubusercontent.com/54267706/165461243-6e6f5ba2-fca5-4cd7-bd2b-5baf5f11f5af.png">

**Comment by Themodem** [comment, marked as answer]
```diff
- "@types/react": "^17.0.0",
+ "@types/react": "^18.0.8",
```

In my case bumping `@types/react` to `^18.0` was enough to resolve this issue

**Reply by iammdsameer** [reply]
what was you react version? for me this also didn't work out.

**Reply by namansamra** [reply]
@Themodem  thank you so much it worked for me.
@itsmdsameerkhan I have react version ^17.0.2

**Reply by cgcab** [reply]
@namansamra or @Themodem by any chance do you happen to find a solution but with react 18.0.0+ ?

**Reply by TShaunik** [reply]
@cgcab I just ran into this while updating our dependencies. From [another issue](https://github.com/gilbarbara/react-inlinesvg/issues/209#issuecomment-1680151085), I bumped Typescript from 4.5.5 to 5.0.2 and it resolved for me. I had tried 5.2.2 but was getting other issues and bumped back down (not sure if that was too high of a version or not); will try again after I finish all the migrations. On React 18.2.0 now.

**Reply by gilbarbara** [reply]
This is probably caused by a [conflict](https://itsjavascript.com/npm-package-cannot-be-used-as-a-jsx-component) in @types/react.
To get things working in the demo, I needed to pin @types/react to 18.0.25 since the VS Code instance uses Typescript 4.x and it can't be changed.

**Reply by cgcab** [reply]
@gilbarbara the @types/react to 18.0.25 also fixed for me the SVG cannot be used ...
![image](https://github.com/gilbarbara/react-inlinesvg/assets/57679996/81ae4914-1a4f-4a91-aa2a-f94913d758b0)

Any chance that the lib gets updated to be more compatible with more up to date version of @types/react ?

**Reply by gilbarbara** [reply]
Hey @cgcab 

Take a look at the link I posted in my previous comment.
This is a problem in the host app, not the library, as it doesn't include either typescript or @types/react in its bundle.
